stdlib/Base64/src/decode.jl

```diff
@@ -5,18 +5,34 @@ const BASE64_CODE_END = 0x40
 const BASE64_CODE_PAD = 0x41
 const BASE64_CODE_IGN = 0x42
 const BASE64_DECODE = fill(BASE64_CODE_IGN, 256)
+const BASE64URL_DECODE = fill(BASE64_CODE_IGN, 256)
 for (i, c) in enumerate(BASE64_ENCODE)
     BASE64_DECODE[Int(c)+1] = UInt8(i - 1)
 end
-BASE64_DECODE[Int(encodepadding())+1] = BASE64_CODE_PAD
-decode(x::UInt8) = @inbounds return BASE64_DECODE[x + 1]
+for (i, c) in enumerate(BASE64URL_ENCODE)
+    BASE64URL_DECODE[Int(c)+1] = UInt8(i - 1)
+end
+
+const PADIND_DECODE = Int(ENCODEPADDING)+1
+BASE64_DECODE[PADIND_DECODE] = BASE64_CODE_PAD
+BASE64URL_DECODE[PADIND_DECODE] = BASE64_CODE_PAD
+
+const TABLE_DECODE = vcat(BASE64_DECODE, BASE64URL_DECODE)
+
+decodeonechar(x::UInt8, decode::Base64Format) = @inbounds return TABLE_DECODE[(x | (UInt16(decode) << 8)) + 1]
+decodeonechar(x::UInt8, decode::UInt8) = @inbounds return TABLE_DECODE[(x | (UInt16(decode) << 8)) + 1]
```

Comment:
This is also related to the problem below, where slight performance regression was observed when giving enum (Base64Format) as an argument. 
Usually it seems to be more simple to pass `decode::Base64Format` as an argument for `decodeonechar`, while method for `decode::UInt8` is also needed when the function `decode_slow` takes `UInt8` instead of `Base64Format` as an argument.

Reply:
Now `decodeonechar(x::UInt8, decode::UInt8)` is removed in 46fed4c.